src/Docker/Service/Version/Validator.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Docker\Service\Version;
+
+use Composer\Semver\Semver;
+use Magento\MagentoCloud\Docker\Service\Config;
+use Magento\MagentoCloud\Package\MagentoVersion;
+use Magento\MagentoCloud\Docker\ConfigurationMismatchException;
+use Magento\MagentoCloud\Package\UndefinedPackageException;
+
+/**
+ * Validate Service versions for docker configuration.
+ */
+class Validator
+{
+    /**
+     * Supported version constraints of services for every Magento version
+     */
+    const MAGENTO_SUPPORTED_SERVICE_VERSIONS = [
+        '<2.2.0' => [
+            Config::KEY_PHP => '7.0.2 || 7.0.4 || ~7.0.6 || ~7.1.0',
+            Config::KEY_DB => '>=10.0 <10.3',
+            Config::KEY_NGINX => '^1.0',
+            Config::KEY_VARNISH => '~3.5 || ~4.0',
+            Config::KEY_REDIS => '~3.2 || ~4.0 || ~5.0',
+            Config::KEY_ELASTICSEARCH => '1.7 || ^2.0',
+            Config::KEY_RABBITMQ => '~3.5',
+        ],
+        '>=2.2.0 <2.2.8' => [
+            Config::KEY_PHP => '~7.0.13 || ~7.1',
+            Config::KEY_DB => '>=10.0 <10.3',
+            Config::KEY_NGINX => '^1.0',
+            Config::KEY_VARNISH => '~4.0 || ~5.0',
+            Config::KEY_REDIS => '~3.2 || ~4.0 || ~5.0',
+            Config::KEY_ELASTICSEARCH => '^2.0 || ^5.0',
```

Comment:
@NadiyaS , also had a question why we use whole major release line in constraints. Currently cloud supports separate defined minor versions of services. For example - for 5.x release line it's 5.2. Not sure if that's fine to leave wide constraints, while all other services are defined in a more accurate way (like MariaDB / Redis, etc.)

Reply:
Discussed offline. Versions were changed.